common/src/main/resources/docker/docker.bom

```diff
@@ -357,6 +365,46 @@ brooklyn.catalog:
             command: |
               cat ${DOCKER_CERT_PATH}/csr.pem
 
+  - id: docker-engine-with-resilience
```

Comment:
Looks good. I compared this with Andrew's version in #329 - it's the same. And I compared it with the deleted version from swarm.bom (the only difference there is the naming: `swarm.recovery.stabilizationDelay` -> `docker.recovery.stabilizationDelay`, and `swarm.recovery.failOnRecurringFailuresInThisDuration` -> `docker.recovery.failOnRecurringFailuresInThisDuration`).